.github/workflows/test-and-lint-php.yml

```diff
@@ -0,0 +1,67 @@
+name: Test & lint PHP codebase
+
+on:
+  pull_request:
+    branches: [ "main" ]
+    types: [opened, synchronize, reopened, ready_for_review]
+    paths:
+      - '**.php'
+      - 'composer.json'
+      - 'composer.lock'
+      - 'phpunit.xml'
+      - 'env.ci'
+
+jobs:
+  test-and-lint-php:
+    name: Test & lint PHP codebase
+    timeout-minutes: 10
+    if: github.event.pull_request.draft == false
+    runs-on: ubuntu-latest
+    services:
+      pgsql:
+        image: postgres:16.1-alpine3.18 #https://hub.docker.com/_/postgres
+        env:
+          POSTGRES_DB: example-app
+          POSTGRES_USER: example-app
+          POSTGRES_PASSWORD: password
+        # Set health checks to wait until postgres has started
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 3s
+          --health-timeout 3s
+          --health-retries 5
+        ports:
+          - 5432:5432
+
+    steps:
+      - uses: actions/checkout@v4.1.1 #https://github.com/actions/checkout
+
+      - name: Validate composer.json and composer.lock
+        working-directory: src
+        run: composer validate
+
+      - name: Cache dependencies
+        uses: actions/cache@v3.3.2 #https://github.com/actions/cache
+        with:
+          path: vendor
+          key: ${{ runner.os }}-composer-dependencies-${{ hashFiles('composer.lock') }}
+          restore-keys: ${{ runner.os }}-composer-dependencies
+
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2 #https://github.com/shivammathur/setup-php
```

Comment:
If we are going to pin versions, lets do it here too

Reply:
with pinned version workflow not working

Reply:
Ok, I can see `2.27.1` is working instead of `v2.27.1`.